### ISSUE TITLE
SSH with ssh forwarding fails with "C:\\HashiCorp\\Vagrant\\embedded\\usr\\bin/ssh.EXE" (Cmder)

### ISSUE THREAD

**Issue by stevenvdr** [hashicorp/vagrant#9101]
I have enabled ssh agent forwarding for my vagrant VM using the provided configuration. However it seems that the ssh.exe provided by vagrant doesn't implement this request correctly. 

I have extracted the ssh command executed from the debug log of vagrant and substituted the vagrant ssh.exe by the default one provided by Cmder. This configuration seems to work correctly.

### Vagrant version
Vagrant 2.0.0

### Host operating system
Windows 10

### Guest operating system
Ubuntu 16.04.3 LTS (GNU/Linux 4.4.0-97-generic x86_64)

### Vagrantfile (relevant config)
config.ssh.forward_agent = true

```ruby
Vagrant.configure("2") do |config|
  config.vm.box = "ubuntu/xenial64"
  config.vm.network "public_network"
  config.ssh.forward_agent = true
end
```

### Other setup
Using cmder (http://cmder.net/). 
ssh-agent is running using the provided configuration in user-profile.cmd of cmder:
```
call "%GIT_INSTALL_ROOT%/cmd/start-ssh-agent.cmd"
```
Confirmed with 'ssh-add -l'.
```
2048 SHA256:... /c/Users/vande/.ssh/id_rsa (RSA)
```
 
### Debug output
https://gist.github.com/stevenvdr/70f06e3b2f94f0bf91bbca11775bdee8

### Expected behavior
When opening the console of the vm using 'vagrant ssh' I expect the output of 'ssh-add -l' to be:
```
$ ssh-add -l
2048 SHA256:... /c/Users/vande/.ssh/id_rsa (RSA)
```

### Actual behavior
```
$ ssh-add -l
Could not open a connection to your authentication agent.
```

**Comment by tdonohue** [comment]
I'm seeing this same behavior using Vagrant 2.0.1 with a host of Windows 10 and guest of Ubuntu 14.04 or 16.04.

With a Vagrantfile similar to above, and an SSH agent running on the host machine:
* `vagrant ssh` fails to perform SSH agent forwarding.  On the guest machine, `ssh-add -l` returns "Could not open a connection to your authentication agent.".  Additionally, no `SSH_AUTH_SOCK` is set.
* However, logging into the same VM via `ssh vagrant@127.0.0.1 -p 2222` (password: vagrant) causes the SSH agent forwarding to work. Then, on the guest machine, `ssh-add -l` shows my forwarded keys, and an `SSH_AUTH_SOCK` is set.

Simply put, `vagrant ssh` on Windows no longer seems to support SSH agent forwarding.  It used to work for me (at some point prior to Vagrant 2.0.0), but no longer works.

**Comment by tdonohue** [comment]
**WORKAROUND:** As long as you have another `ssh` tool in your Windows `Path` (e.g. the ssh.exe provided by Git for Windows or GitHub Desktop), then you can set the `VAGRANT_PREFER_SYSTEM_BIN` environment variable, and Vagrant will use whatever `ssh` it finds on your path first.

Discovered this via https://github.com/hashicorp/vagrant/issues/8739#issuecomment-343958840

More in the docs at https://www.vagrantup.com/docs/other/environmental-variables.html#vagrant_prefer_system_bin

After setting `VAGRANT_PREFER_SYSTEM_BIN=true` locally, now `vagrant ssh` works again!

**Comment by chrisroberts** [comment]
Closing this up as the environment variable mentioned in this comment (https://github.com/hashicorp/vagrant/issues/9101#issuecomment-360265849) will resolve the issue.

Cheers!
